lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java

```diff
@@ -1798,7 +1798,8 @@ private void renderAndDrawAsBitmap(Canvas originalCanvas, CompositionLayer compo
     int renderWidth = (int) Math.ceil(softwareRenderingTransformedBounds.width());
     int renderHeight = (int) Math.ceil(softwareRenderingTransformedBounds.height());
 
-    if (renderWidth <= 0 || renderHeight <= 0) {
+    // Safeguard against errors during Bitmap creation by returning early if dimensions are invalid.
+    if (renderWidth <= 0 || renderHeight <= 0 || renderWidth > bounds.width() || renderHeight > bounds.height()) {
```

Comment:
Wouldn't this not render an image if it scaled up so it covered the whole screen + 1px?

Reply:
In that case can we clip the bounds so that we only allocate a bitmap large enough to draw what's shown onscreen?
I'm trying to guard against cases where the Canvas matrix returns an invalid scale or translation that causes something like a 1,000,000 x 1,000,000 bitmap to be allocated.